File: expense-tracker-frontend/src/Service/transactionService.js
import api from './api';

// ✅ Get all transactions
export const getAllTransactions = () => api.get('/transactions');

// ✅ Get transaction by ID
export const getTransactionById = (id) => api.get(`/transactions/${id}`);

// ✅ Get transactions by user name
export const getTransactionsByUserName = (userName) =>
  api.get('/transactions/by-user', { params: { userName } });

// ✅ Get transactions by date (pass date string in YYYY-MM-DD format)
export const getTransactionsByDate = (date) =>
  api.get('/transactions/by-date', { params: { date } });

// ✅ Get transactions by category (enum, e.g., FOOD)
export const getTransactionsByCategory = (category) =>
  api.get('/transactions/by-category', { params: { category } });

// ✅ Get transactions by type (enum, e.g., INCOME or EXPENSE)
export const getTransactionsByType = (type) =>
  api.get('/transactions/by-type', { params: { type } });

// ✅ Create a new transaction
export const createTransaction = (data) => api.post('/transactions', data);

// ✅ Update an existing transaction by ID
export const updateTransaction = (id, data) => api.put(`/transactions/${id}`, data);

// ✅ Delete a transaction by ID
export const deleteTransaction = (id) => api.delete(`/transactions/${id}`);

File: Personal-Expense-Tracker/src/main/java/com/example/Personal/Expense/Tracker/Service/TransactionService.java
package com.example.Personal.Expense.Tracker.Service;

import com.example.Personal.Expense.Tracker.Dto.TransactionDto;
import com.example.Personal.Expense.Tracker.Entity.Category;
import com.example.Personal.Expense.Tracker.Entity.Type;
import org.springframework.stereotype.Service;

import java.time.LocalDate;
import java.util.List;



public interface TransactionService {

    TransactionDto createTransaction(TransactionDto transactionDto);
    TransactionDto getTransactionById(Long id);
    List<TransactionDto> getAllTransactions();
    List<TransactionDto> getTransactionsByDate(LocalDate date);
    List<TransactionDto> getTransactionByCategory(Category category);
    List<TransactionDto> getTransactionsByType(Type type);
    TransactionDto updateTransaction(Long id,TransactionDto transactionDto);
    List<TransactionDto> getTransactionsByUserName(String userName);
    void deleteTransaction(Long id);



}
